src/main/java/org/testng/JarFileUtils.java

```diff
@@ -27,7 +33,8 @@
     JarFileUtils(IPostProcessor processor, String xmlPathInJar, List<String> testNames) {
         this.processor = processor;
         this.xmlPathInJar = xmlPathInJar;
-        this.testNames = testNames;
+        //Ensure that dynamic list operations supported
+        this.testNames = (testNames == null ? null : new ArrayList<>(testNames));
```

Comment:
You can revert this change because you don't need it in this specific class.

Reply:
yes